src/streamlink_cli/output.py

```diff
@@ -108,12 +109,27 @@ def _create_arguments(self):
             filename = self.http.url
         else:
             filename = "-"
-
         args = self.args.format(filename=filename)
         cmd = self.cmd
+        
+        if self.title is not None:
+            #vlc
+            if self.cmd in SUPPORTED_PLAYERS['vlc']:
+                self.title.replace("$","$$").replace("\$$","$") #see https://wiki.videolan.org/Documentation:Format_String/, allow escaping with \$
+                if is_win32:
+                    return cmd + " --input-title-format=\"" + self.title + "\" " + args
+                return shlex.split(cmd) + shlex.split("--input-title-format=" + shlex_quote(self.title)) + shlex.split(args)
```

Comment:
Here is the real meat of the issue... On macOS and Linux it works great to let `list2cmdline` deal with the escaping (as it should). 

```python
return shlex.split(cmd) + ["--input-title-format", self.title] + shlex.split(args)
```

But on Windows we don't want to apply `shlex.split`, as it doesn't work quite right. 

Doing this works for me on Windows, Linux and macOS. 
```python
args = self.args.format(filename=filename)
cmd = self.cmd
extra_args = []

if self.title is not None:
    # vlc
    if self.cmd in SUPPORTED_PLAYERS['vlc']:
        # see https://wiki.videolan.org/Documentation:Format_String/, allow escaping with \$
        self.title.replace("$", "$$").replace("\$$", "$")
        extra_args.extend(["--input-title-format", self.title])

    # mpv
    if self.cmd in SUPPORTED_PLAYERS['mpv']:
        # see https://mpv.io/manual/stable/#property-expansion, allow escaping with \$
        self.title.replace("$", "$$").replace("\$$", "$")
        extra_args.extend(["--title", self.title])

# other player
if is_win32:
    return cmd + " " + subprocess.list2cmdline(extra_args) + " " + args
return shlex.split(cmd) + extra_args + shlex.split(args)
```

Reply:
Yeah, that looks better to me. I think I had it in my head that win32 is already protected from title-based attacks somewhere down the line with how the player is launched, but your code looks cleaner and uses list2cmdline, so I'll use it.